File: pe/003/euler3.swift
func getFactors(max : Int64) -> [Int64] {
	var num : Int64 = max 
	var factors : [Int64] = []
	for var i : Int64 = 2; i <= num; i++ {
		while num % i == 0 {
			factors.append(i)
			num = num / i
		}
	}
	return factors
}

var arr = getFactors(600851475143)
var lastIdx = arr.count-1;
println(arr)
println(arr[lastIdx])

File: pe/003/euler3.js
function getFactors(num) {
	var factors = [];
	for (var i = 2; i <= num; i++) {
		while (num % i === 0) {
			factors.push(i);
			num = num / i;
		}

	}
	return factors;
}

var f = getFactors(600851475143);
console.log(f[f.length-1]);

